integrationTests/chainSimulator/bridge/simpleEsdt_test.go

```diff
@@ -0,0 +1,446 @@
+	txData = createNftArgs("da3-NFT3-gew3gr", uint64(7), big.NewInt(2), core.DynamicNFT, creatorAddress)
+	executeSimpleEsdtOperationWithError(t, txData, expectedFullError)
+}
+
+func TestSovereignChainSimulator_CreateNftWithInvalidTypeShouldFail(t *testing.T) {
+	if testing.Short() {
+		t.Skip("this is not a short test")
+	}
+
+	expectedError := "invalid arguments to process built-in function, invalid esdt type"
+
+	expectedFullError := fmt.Sprintf("%s %d (%s)", expectedError, core.Fungible, core.Fungible.String())
+	txData := createNftArgs("da-TKN-ten731", uint64(0), big.NewInt(1), core.Fungible, creatorAddress)
+	executeSimpleEsdtOperationWithError(t, txData, expectedFullError)
+
+	expectedFullError = fmt.Sprintf("%s %d (%s)", expectedError, core.NonFungible, core.NonFungible.String())
+	txData = createNftArgs("da-NFT-4g4325", uint64(7), big.NewInt(1), core.NonFungible, creatorAddress)
+	executeSimpleEsdtOperationWithError(t, txData, expectedFullError)
+
+	expectedFullError = fmt.Sprintf("%s %d (%s)", expectedError, core.ESDTType(8), core.ESDTType(8).String())
+	txData = createNftArgs("da-NFT-ten731", uint64(7), big.NewInt(1), core.ESDTType(8), creatorAddress)
+	executeSimpleEsdtOperationWithError(t, txData, expectedFullError)
+}
+
+func executeSimpleEsdtOperationWithError(
+	t *testing.T,
+	txData string,
+	expectedErr string,
+) {
+	roundsPerEpoch := core.OptionalUint64{
+		HasValue: true,
+		Value:    20,
+	}
+
+	whiteListedAddress := "erd1qqqqqqqqqqqqqpgqmzzm05jeav6d5qvna0q2pmcllelkz8xddz3syjszx5"
+	cs, err := chainSimulator.NewChainSimulator(chainSimulator.ArgsChainSimulator{
+		BypassTxSignatureCheck:   true,
+		TempDir:                  t.TempDir(),
+		PathToInitialConfig:      defaultPathToInitialConfig,
+		NumOfShards:              3,
+		GenesisTimestamp:         time.Now().Unix(),
+		RoundDurationInMillis:    uint64(6000),
+		RoundsPerEpoch:           roundsPerEpoch,
+		ApiInterface:             api.NewNoApiInterface(),
+		MinNodesPerShard:         3,
+		MetaChainMinNodes:        3,
+		NumNodesWaitingListMeta:  0,
+		NumNodesWaitingListShard: 0,
+		AlterConfigsFunction: func(cfg *config.Configs) {
+			cfg.GeneralConfig.VirtualMachine.Execution.TransferAndExecuteByUserAddresses = []string{whiteListedAddress}
+			cfg.EpochConfig.EnableEpochs.DynamicESDTEnableEpoch = 0
+		},
+	})
+	require.Nil(t, err)
+	require.NotNil(t, cs)
+
+	defer cs.Close()
+
+	err = cs.GenerateBlocksUntilEpochIsReached(4)
+	require.Nil(t, err)
+
+	nodeHandler := cs.GetNodeHandler(0)
+	systemContractDeploy := chainSim.GetSysContactDeployAddressBytes(t, nodeHandler)
+
+	initialAddress := "erd1l6xt0rqlyzw56a3k8xwwshq2dcjwy3q9cppucvqsmdyw8r98dz3sae0kxl"
+	initialAddrBytes, err := nodeHandler.GetCoreComponents().AddressPubKeyConverter().Decode(initialAddress)
+	err = cs.SetStateMultiple([]*dtos.AddressState{
+		{
+			Address: initialAddress,
+			Balance: "10000000000000000000000",
+		},
+		{
+			Address: esdtSystemAccount, // init sys account
```

Comment:
Could we use some existing funcs for this here, you you used above  `initOwnerAndSysAccState` ?

Reply:
I made a common func for this init